src/skydb.ts

```diff
@@ -1,105 +1,65 @@
-import { RegistryValue, SignedRegistryValue } from "./registry";
-import { promiseTimeout, trimUriPrefix, uriSkynetPrefix } from "./utils";
+import { HashRegistryEntry, PublicKey, SecretKey } from "./crypto";
+import { RegistryEntry } from "./registry";
+import { trimUriPrefix, uriSkynetPrefix } from "./utils";
 
-// FILEID_V1 represents version 1 of the FileID object
-export const FILEID_V1 = 1;
-
-// FileType is the type of the file
-export enum FileType {
-  Invalid, // 0 is invalid
-  PublicUnencrypted,
-}
-
-// getFile will lookup the entry for given skappID and filename, if it exists it
-// will try and download the file behind the skylink it has found in the entry.
-export async function getFile(this: SkynetClient, user: User, fileID: FileID): Promise<SkyFile> {
+export async function getJSON(this: SkynetClient, publicKey: PublicKey, dataKey: string): Promise<{ json: object, revision: number }|null> {
   // lookup the registry entry
-  const existing = await this.lookupRegistry(user, fileID);
-  if (!existing) {
+  const entry = await this.registry.lookup(publicKey, dataKey);
+  if (entry === null) {
     throw new Error("not found");
   }
 
-  // download the data in that Skylink
-  const skylink = existing.value.data;
+  // Download the data in that Skylink
+  // TODO: Replace with download request method.
+  const skylink = entry.value.data;
   const response = await this.executeRequest({
     ...this.customOptions,
     method: "get",
     url: this.getSkylinkUrl(skylink),
   });
 
-  // wrap the data in a skyfile and return it
-  const metadata = JSON.parse(response.headers["skynet-file-metadata"]);
-  const file = new SkyFile(new File([response.data], metadata.filename, { type: response.headers["content-type"] }));
-  return file;
+  return { json: response.data, revision: entry.revision };
 }
 
-// setFile uploads a file and sets updates the registry
-export async function setFile(this: SkynetClient, user: User, fileID: FileID, file: SkyFile): Promise<boolean> {
-  // upload the file to acquire its skylink
-  const customFilename = fileID.filename;
-  const skylink = await this.uploadFile(file.file, { customFilename });
-
-  // fetch the current value to find out the revision, timeout after 2s
-  let existing: SignedRegistryValue | null;
-  try {
-    existing = await promiseTimeout(this.lookupRegistry(user, fileID), 2000);
-  } catch (error) {
-    existing = null;
+export async function setJSON(this: SkynetClient, privateKey: SecretKey, dataKey: string, json: object, revision: number = -1): Promise<boolean> {
+  // Upload the data to acquire its skylink
+  // TODO: Replace with upload request method.
+  const response = await this.executeRequest({
+    ...this.customOptions,
+    method: "post",
+    endpointPath: "/skynet/skyfile",
+    data: json.toString(),
+  });
+  const skylink = response.skylink;
+
+  const publicKey = pki.ed25519.publicKeyFromPrivateKey({privateKey});
+  if (revision === -1) {
```

Comment:
did we land on this behavior? (works for me)

Reply:
the reason I ask is because I guess the `-1` convention might be considered somewhat of a code smell, even though I don't mind that type of thing

Reply:
Dunno, I got it from your proposal. I think it's fine as a default value but maybe it's idiosyncratic for JS.